### PR TITLE
Add missing RBI definitions for Array

### PR DESCRIPTION
**Pull request by Morriar**
No signatures.

### CODE REVIEW

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ https://go/builds/bui_HOW0LIBXk7qqQy

File: rbi/core/array.rbi [hunk 1]
```diff
@@ -2460,6 +2505,23 @@ class Array < Object
...
+  # ```
+  #
+  # If a block is given, the results of the block on each element of the array
+  # will be used as pairs.
+  #
+  # ```ruby
+  # ["foo", "bar"].to_h {|s| [s.ord, s]}
+  #   # => {102=>"foo", 98=>"bar"}
+  # ```
+  def to_h(&blk); end
```
Comment (elliottt) [inline rbi/core/array.rbi:2523]: This is overriding the definition on `Enumerable`, and causing some errors in the stripe codebase. What do you think about just removing this one?

**Comment by Morriar** [comment]
@elliottt I removed the definition for `to_h` both in the RBI and expected files.

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ https://go/builds/bui_HOWiKOdvcYpHqI

**Pull request merged**